Eshop.WebApi/Controllers/CategoryController.cs

```diff
@@ -23,33 +22,34 @@ public List<Category> GetCategory()
         [HttpGet("{itemType}")]
         public Category GetCategory(string itemType) 
         { 
-            return Categories.First(x => x.ItemType == itemType);
+            return Categories.First(x => x.categoryName == itemType);
         }
 
         [HttpPost]
-        public Category CreateProduct(string itemType) 
+        public Category CreateCategory(string categoryName) 
         { 
-            var newItemType = Categories.Max(x => x.ItemType) + 1;
-            var newCategory = new Category(newItemType);
+            var newCategoryName = Categories.Max(x => x.categoryName) + 1;
```

Comment:
This is whole wrong. What did I say? That you should take and inspiration from the ProductController.cs
You even have it in the Story/Task as description. So you need to read and follow the instruction that boss told you not doing whatever you think is right. Rework it.